src/main/java/seedu/address/model/particpant/Participant.java

```diff
@@ -0,0 +1,213 @@
+     */
+    public Participant(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
+        super(name, phone, email, address, tags);
+    }
+
+    /**
+     * Builder chaining method for retuning participant object with certain birthdate.
+     * This is to allow the use of Person template along with Person but adding extension fields.
+     * i.e. new Participant(name, phone, email, address, tags).withBirthDate(year, month, dayOfMonth)
+     *
+     * @param year       this year or before
+     * @param month      month of the year specified
+     * @param dayOfMonth day of the month specified
+     * @return another Participant with same particulars object but with birth date
+     */
+    public Participant withBirthDate(int year, int month, int dayOfMonth) {
+        Participant participantToReturn = new Participant(getName(), getPhone(), getEmail(), getAddress(), getTags());
+        participantToReturn.nextOfKins = nextOfKins;
+        participantToReturn.notes = notes;
+        // set birth Date
+        participantToReturn.birthDate = BirthDate.of(year, month, dayOfMonth);
+
+        return participantToReturn;
+    }
+
+    /**
+     * Builder chaining method for retuning participant object with certain birthdate.
+     * i.e. new Participant(name, phone, email, address, tags).withBirthDate(year, month, dayOfMonth).withNotes(notes)
+     * This chaining method is not required unless the participants were to be set with certain notes set
+     *
+     * @param notes set of Note object
+     * @return this Participant but with new notes
+     */
+    public Participant withNotes(Set<Note> notes) {
+        Participant participantToReturn = new Participant(getName(), getPhone(), getEmail(), getAddress(), getTags());
+        participantToReturn.nextOfKins = nextOfKins;
+        participantToReturn.birthDate = birthDate;
+
+        // set notes
+        participantToReturn.notes = notes;
+
+        return participantToReturn;
+    }
+
+
+    /**
+     * Builder chaining method for retuning participant object with certain birthdate.
+     * i.e. new Participant(name, phone, email, address, tags).withBirthDate(year, month, dayOfMonth).withNotes(notes)
+     * This chaining method is not required unless the participants were to be set with certain next of kins arrayList
+     *
+     * @param nextOfKins array list of next of kins
+     * @return this Participant but with new nextOfKins
+     */
+    public Participant withNextOfKins(ArrayList<Person> nextOfKins) {
+        Participant participantToReturn = new Participant(getName(), getPhone(), getEmail(), getAddress(), getTags());
+        participantToReturn.notes = notes;
+        participantToReturn.birthDate = birthDate;
+
+        // set next of kins
+        participantToReturn.nextOfKins = nextOfKins;
+
+        return participantToReturn;
+    }
+
+    /**
+     * @return this object's birthDate
+     */
+    public BirthDate getBirthDate() {
```

Comment:
I'm not sure if all these getter methods are necessary at this point. I think they can be removed first and then added on in subsequent commits if they are really needed. I'm cautious about whether these getters violate abstraction.

Reply:
getter is used in EditCommand so I think will definitely be used to create new objects and also to get attributes in general